verifier-sdk/src/light_transaction.rs

```diff
@@ -1151,3 +1142,80 @@ impl<
         }
     }
 }
+
+#[cfg(all(target_os = "solana", feature = "custom-heap"))]
+pub mod custom_heap {
+    use anchor_lang::{
+        prelude::*,
+        solana_program::entrypoint::{HEAP_LENGTH, HEAP_START_ADDRESS},
+    };
+    #[global_allocator]
+    static A: BumpAllocator = BumpAllocator {
+        start: HEAP_START_ADDRESS as usize,
+        len: HEAP_LENGTH,
+    };
+
+    use std::{alloc::Layout, mem::size_of, ptr::null_mut, usize};
+
+    pub struct BumpAllocator {
+        pub start: usize,
+        pub len: usize,
+    }
+
+    impl BumpAllocator {
+        const RESERVED_MEM: usize = 1 * size_of::<*mut u8>();
+
+        /// Return heap position as of this call
+        pub unsafe fn pos(&self) -> usize {
+            let pos_ptr = self.start as *mut usize;
+            *pos_ptr
+        }
+
+        /// Reset heap start cursor to position.
+        /// ### This is very unsafe, use wisely
```

Comment:
The convention is to use `/// # Safety` for `unsafe` functions in docstrings. So something like:

```suggestion
        /// Reset heap start cursor to position `pos`.
        ///
        /// # Safety
        ///
        /// This is very unsafe and forbidden to use outside of the custom
        /// allocator context. When used incorrectly, it can override an incorrect
        /// chunk of memory.
```